In [5]:
from tensorflow.keras import datasets, layers, models

In [7]:
(train_images, train_labels), (test_images, test_labels) = datasets.mnist.load_data()

train_images, test_images = train_images/255.0 , test_images/255.0

11493376/11490434 [==============================] - 0s 0us/step


In [24]:
model=models.Sequential()
model.add(layers.Flatten(input_shape=(28,28)))
model.add(layers.Dense(20, activation='relu'))
model.add(layers.Dense(10,activation='softmax'))
model.summary()

Model: "sequential_8"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
flatten_3 (Flatten)          (None, 784)               0         
_________________________________________________________________
dense_10 (Dense)             (None, 20)                15700     
_________________________________________________________________
dense_11 (Dense)             (None, 10)                210       
Total params: 15,910
Trainable params: 15,910
Non-trainable params: 0
_________________________________________________________________


In [30]:
model.compile(optimizer='adam',loss='sparse_categorical_crossentropy',metrics=['accuracy'])
model.fit(train_images,train_labels,epochs=10)

Epoch 1/10
1875/1875 [==============================] - 4s 2ms/step - loss: 0.1403 - accuracy: 0.9591
Epoch 2/10
1875/1875 [==============================] - 4s 2ms/step - loss: 0.1320 - accuracy: 0.9615
Epoch 3/10
1875/1875 [==============================] - 4s 2ms/step - loss: 0.1251 - accuracy: 0.9636
Epoch 4/10
1875/1875 [==============================] - 4s 2ms/step - loss: 0.1192 - accuracy: 0.9650
Epoch 5/10
1875/1875 [==============================] - 4s 2ms/step - loss: 0.1144 - accuracy: 0.9662
Epoch 6/10
1875/1875 [==============================] - 4s 2ms/step - loss: 0.1095 - accuracy: 0.9680
Epoch 7/10
1875/1875 [==============================] - 4s 2ms/step - loss: 0.1066 - accuracy: 0.9684
Epoch 8/10
1875/1875 [==============================] - 4s 2ms/step - loss: 0.1014 - accuracy: 0.9693
Epoch 9/10
1875/1875 [==============================] - 4s 2ms/step - loss: 0.0971 - accuracy: 0.9711
Epoch 10/10
1875/1875 [==============================] - 4s 2ms/step - loss: 0.095

In [32]:
test_acc = model.evaluate(test_images, test_labels)

print('\nTest accuracy:', test_acc)

313/313 [==============================] - 1s 2ms/step - loss: 0.1400 - accuracy: 0.9598

Test accuracy: [0.1400364190340042, 0.9598000049591064]


<u>**MNIST Feed Forward Network**</u>


Training Accuracy: 0.9710

Testing Accuracy: 0.9598